models/victoriametrics.go

```diff
@@ -0,0 +1,124 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package models
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+
+	config "github.com/percona/promconfig"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+
+	// enables cache for vmdb responses.
+	vmCacheEnableEnv = "VM_CACHE_ENABLE"
+	// enables victoriametrics services at supervisor config.
+	vmTestEnableEnv = "VM_TEST_ENABLE"
+)
+
+// VictoriaMetricsParams - defines flags and settings for victoriametrics.
+type VictoriaMetricsParams struct {
+	// VMDBFlags defines additional flags for VictoriaMetrics DB.
+	VMDBFlags []string
+	// VMAlertFlags additional flags for VMAlert.
+	VMAlertFlags []string
+	// Enables VictoriaMetrics.
+	Enabled bool
+	// BaseConfigPath defines path for basic prometheus config.
+	BaseConfigPath string
+}
+
+// NewVictoriaMetricsParams - returns configuration params for VictoriaMetrics.
+func NewVictoriaMetricsParams(basePath string) (*VictoriaMetricsParams, error) {
+	vmp := &VictoriaMetricsParams{
+		BaseConfigPath: basePath,
+	}
+	if enabledVar := os.Getenv(vmTestEnableEnv); enabledVar != "" {
+		parsedBool, err := strconv.ParseBool(enabledVar)
+		if err != nil {
+			return vmp, errors.Wrapf(err, "cannot parse %s as bool", vmCacheEnableEnv)
+		}
+		vmp.Enabled = parsedBool
+	}
+
+	if err := vmp.UpdateParams(); err != nil {
+		return vmp, err
+	}
+
+	return vmp, nil
+}
+
+// UpdateParams - reads configuration file and updates corresponding flags.
+func (vmp *VictoriaMetricsParams) UpdateParams() error {
+	if err := vmp.loadVMAlertParams(); err != nil {
+		return errors.Wrap(err, "cannot update VMAlertFlags config param")
+	}
+	if err := vmp.loadVMDBParams(); err != nil {
+		return errors.Wrap(err, "cannot update VMDBFlags config param")
+	}
+
+	return nil
+}
+
+// loadVMAlertParams - load params and converts it to vmalert flags.
+func (vmp *VictoriaMetricsParams) loadVMAlertParams() error {
+	buf, err := ioutil.ReadFile(vmp.BaseConfigPath)
```

Comment:
Why do we get flags from config file? Was this discussed?

Reply:
Coz, user may define own rule files path with prometheus.yaml, mounted to container. And we have to pass this additional rules to `vmalert` supervisor config.